### ISSUE TITLE
VS 2022 dropped declaration for off_t

### ISSUE THREAD

**Issue by bscottm** [openssl/openssl#18310]
Encountered this compile error while building `pcap` on Windows 10 with OpenSSL 3.0.3 via the `cmake` builder (which shouldn't really matter much, but...):

```
 Microsoft (R) Build Engine version 17.1.0+ae57d105c for .NET Framework
  Copyright (C) Microsoft Corporation. All rights reserved.

  Microsoft (R) Build Engine version 17.1.0+ae57d105c for .NET Framework
  Copyright (C) Microsoft Corporation. All rights reserved.

    Checking Build System
    Generating grammar.c, grammar.h
    Generating scanner.c, scanner.h
    Building Custom Rule C:/Users/bsm21317/play/simh/cmake/build-vs2022/pcap-dep-prefix/src/pcap-dep/CMakeLists.txt
    Building Custom Rule C:/Users/bsm21317/play/simh/cmake/build-vs2022/pcap-dep-prefix/src/pcap-dep/CMakeLists.txt
    bpf_dump.c
    bpf_filter.c
    bpf_image.c
    etherent.c
    fmtutils.c
    gencode.c
    nametoaddr.c
    optimize.c
    pcap-common.c
    pcap.c
    savefile.c
    sf-pcapng.c
    sf-pcap.c
    charconv.c
    win_asprintf.c
    pcap-npf.c
    pcap-new.c
C:\...\include\openssl/ssl.h(1970,56): error C2146: syntax error: missing ')' before identifier 'offset' [ C:\...\pcap-dep-prefix\src\pcap-dep-build\pcap_static.vcxproj] [C:\...\simh\cmake\build- vs2022\pcap-dep.vcxproj]
C:\...\include\openssl/ssl.h(1970,56): error C2081: 'off_t': name in formal parameter list illegal [C:\...\pcap-dep-prefix\src\pcap-dep-build\pcap_static.vcxproj] [C:\...\simh\cmake\build-vs2022 \pcap-dep.vcxproj]
```

More lines similar to the two compile errors follow, which I've elided out. The solution seems to be something like:

```
#if defined(_MSC_VER) && _MSC_VER >= 1930
typedef _off_t off_t;
#endif
```

I know the code base tries to avoid platform-specific foo insofar as feasible, so I'm not entirely sure where to place it as a proposed fix.

**Comment by t8m** [comment]
All the use of off_t is associated with ktls which is clearly not supported on Windows. We could probably just `#ifdef OPENSSL_NO_KTLS` it except for that we have the SSL_sendfile() call, which also does not work without ktls but the API call exists and is present currently even with OPENSSL_NO_KTLS.

@openssl/otc any opinion on how to fix this?

**Comment by jakirkham** [comment]
Just wanted to check in on the status here, we are trying to rebuild packages for OpenSSL 3 to move to that version. However this issue tripped us up.

**Comment by mattcaswell** [comment]
> any opinion on how to fix this?

Could we somehow disable SSL_sendfile() only on Windows? Or only for appropriate versions of `_MSC_VER`?

**Comment by traversaro** [comment]
While the proper solution at the openssl level is to ensure that `SSL_sendfile` is not enabled on Windows, if anyone needs a fix for pcap it is possible to find it at https://github.com/conda-forge/libpcap-feedstock/pull/18#issuecomment-1535475543, in particular the patch https://github.com/conda-forge/libpcap-feedstock/commit/331f347cae812d082916cb301042c041220112d1 .

**Comment by nhorman** [comment]
Marking as inactive, to be closed when 3.4 dev cycle ends, barring further input
